workshops/accelerating_inference_with_openvino_and_pytorch/llm_chatbot/llm_chatbot_itdc.ipynb

```diff
@@ -0,0 +1,1090 @@
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "\n",
+    "os.environ[\"GIT_CLONE_PROTECTION_ACTIVE\"] = \"false\"\n",
+    "\n",
+    "!{sys.executable} -m pip install -Uq pip\n",
+    "!{sys.executable} -m pip uninstall -q -y optimum optimum-intel openvino transformers\n",
+    "!{sys.executable} -m pip install --pre -Uq \"openvino>=2024.2.0\" openvino-tokenizers[transformers] --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly\n",
+    "!{sys.executable} -m pip install -q --extra-index-url https://download.pytorch.org/whl/cpu\\\n",
+    "\"git+https://github.com/openvinotoolkit/nncf.git\"\\\n",
+    "\"torch>=2.1.1\"\\\n",
+    "\"datasets\" \\\n",
+    "\"accelerate\"\\\n",
+    "\"onnx\" \"einops\" \"transformers_stream_generator\" \"tiktoken\" \"transformers>=4.40\" \"bitsandbytes\"\n",
+    "!{sys.executable} -m pip install -q optimum-intel\n",
+    "!{sys.executable} -m pip install -q \"transformers==4.42.4\" --extra-index-url https://download.pytorch.org/whl/cpu"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b8c1d0b2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "username = os.environ.get('USER')\n",
+    "user_bin_path = os.path.expanduser(f\"/home/{username}/.local/bin\")\n",
+    "sys.path.append(user_bin_path)\n",
+    "print(sys.path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "f39ca954-61d2-45c5-a7f9-7fce1acc277f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "LLM config will be updated\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "from pathlib import Path\n",
+    "import requests\n",
+    "import shutil\n",
+    "\n",
+    "# fetch model configuration\n",
+    "\n",
+    "config_shared_path = Path(\"../../utils/llm_config.py\")\n",
+    "config_dst_path = Path(\"llm_config.py\")\n",
+    "\n",
+    "if not config_dst_path.exists():\n",
+    "    if config_shared_path.exists():\n",
+    "        try:\n",
+    "            os.symlink(config_shared_path, config_dst_path)\n",
+    "        except Exception:\n",
+    "            shutil.copy(config_shared_path, config_dst_path)\n",
+    "    else:\n",
+    "        r = requests.get(url=\"https://raw.githubusercontent.com/openvinotoolkit/openvino_notebooks/latest/utils/llm_config.py\")\n",
+    "        with open(\"llm_config.py\", \"w\", encoding=\"utf-8\") as f:\n",
+    "            f.write(r.text)\n",
+    "elif not os.path.islink(config_dst_path):\n",
+    "    print(\"LLM config will be updated\")\n",
+    "    if config_shared_path.exists():\n",
+    "        shutil.copy(config_shared_path, config_dst_path)\n",
+    "    else:\n",
+    "        r = requests.get(url=\"https://raw.githubusercontent.com/openvinotoolkit/openvino_notebooks/latest/utils/llm_config.py\")\n",
```

Comment:
Do you need to download anything if you provide the file here? If so, the file will be downloaded from the notebooks repository and the content may change unexpectedly...